src/python/pants/engine/internals/native_engine.pyi

```diff
@@ -978,6 +979,11 @@ class PyRemotingOptions:
 class PyLocalStoreOptions:
     def __init__(self, **kwargs: Any) -> None: ...
 
+class PyIntrinsicsOptions:
+    def __init__(
+        self, *, downloads_intrinsic_error_delay: timedelta, downloads_intrinsic_max_retries: int
+    ) -> None: ...
```

Comment:
The infrastructure required to plumb these options through for the download intrinsics seems quite awkward. Another option might be passing those args to the download intrinsic's input, i.e. augment: https://github.com/pantsbuild/pants/blob/fa583d6b36df4c7bb1c560dc5429d1bfb474ba51/src/python/pants/engine/fs.py#L257-L274


Potentially either have the two callers (`download_file` and `download_from_s3`) pull the option out of global options or have a yet-another-rule that does that.

What do you think?

Reply:
> Potentially either have the two callers (download_file and download_from_s3) pull the option out of global options or have a yet-another-rule that does hydrates the simple download request into the "real" value that's passed to the intrinsic with the additional retry info.

That's a good idea! The existing `NativeDownloadFile` type can carry the retry behavior options. And I avoid all of the plumbing for those options into the `Core` type.

Reply:
The latest commits implement this approach. Much cleaner! Thanks for the idea!